src/test/FuzzerImpl.cpp

```diff
@@ -688,17 +763,87 @@ TransactionFuzzer::initialize()
         genOffersForPair(ASSET_B, ASSET_C, {13, 2, 14, 3, 15}, ltx);
         genOffersForPair(ASSET_C, ASSET_D, {6, 3, 7, 4, 8}, ltx);
 
-        // construct transaction
-        auto tx = createFuzzTransactionFrame(mSourceAccountID, ops,
-                                             mApp->getNetworkID());
+        attemptToApplyOps(ltx, mSourceAccountID, ops, *mApp);
+
+        ltx.commit();
+    }
+
+    {
+        LedgerTxn ltx(ltxOuter);
 
-        tx->attemptApplication(*mApp, ltx);
+        xdr::xvector<Operation> ops;
 
-        if (tx->getResultCode() != txSUCCESS)
+        // Reduce account balances so that fuzzing has a better change
+        // of exercising limits.
+        for (uint8_t i = 0; i < FuzzUtils::NUMBER_OF_PREGENERATED_ACCOUNTS; ++i)
         {
-            throw std::runtime_error("Error while initializing ledger state");
+            PublicKey account;
+            FuzzUtils::setShortKey(account, i);
+
+            // Reduce "account"'s native balance by paying the root.
+            LedgerKey accountKey;
+            accountKey.type(ACCOUNT);
+            accountKey.account().accountID = account;
+            LedgerEntry accountEntry = ltx.load(accountKey).current();
```

Comment:
This is quite unidiomatic for a few reasons:
1. `accountEntry` does not have type `AccountEntry` so the name is misleading
2. You copied out of a `LedgerTxnEntry` instead of just using `current` when you need it. This is bypassing some of the `LedgerTxn` safety guarantees. Don't take references to or copy from `current`, unless there is some deep necessity.

Reply:
Done (along with using `loadAccount()`).